maintenance_request_time_performance/views/maintenance_views.xml

```diff
@@ -0,0 +1,69 @@
+<odoo>
+<data>
+    <!-- New tree view: stages - added in_progress boolean -->
+    <record id="hr_equipment_stage_view_tree" model="ir.ui.view">
+        <field name="name">equipment.stage.tree</field>
+        <field name="model">maintenance.stage</field>
+        <field name="inherit_id" ref="maintenance.hr_equipment_stage_view_tree"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='done']" position="before">
+                <field name="in_progress"/>
+            </xpath>
+        </field>
+    </record>
+
+    <!-- New form view: MaintenanceRequest - added start and end datetime -->
+    <record id="hr_equipment_request_view_form" model="ir.ui.view">
+        <field name="model">maintenance.request</field>
+        <field name="inherit_id" ref="maintenance.hr_equipment_request_view_form"/>
+        <field name="priority" eval="100"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='priority']" position="after">
+                <field name="start_request_datetime_required" invisible="1"/>
+                <field name="end_request_datetime_required" invisible="1"/>
+                <field name="start_request_datetime"
+                       attrs="{'required': [('start_request_datetime_required', '=', True)]}"/>
+                <field name="end_request_datetime"
+                       attrs="{'required': [('end_request_datetime_required', '=', True)]}"/>
+                <label for="real_request_duration"/>
+                <div attrs="{'invisible': ['|', ('start_request_datetime', '=', False), ('end_request_datetime', '=', False)]}">
+                    <field name="real_request_duration" widget="float_time" class="oe_inline"/> <span class="ml8">hours</span>
+                </div>
+                <label for="completion_delay"/>
+                <div attrs="{'invisible': ['|', ('schedule_date', '=', False), ('end_request_datetime', '=', False)]}">
+                    <field name="completion_delay" widget="float_time" class="oe_inline"/> <span class="ml8">hours</span>
+                </div>
```

Comment:
Should `completion_delay` be visible for everyone in the `form` view, or not? I'm not sure...

Reply:
Completion_delay is visible for everyone according to user permissions, but it is invisible when (now) end_request_datetime is False

Reply:
I am unable to check end_request_datetime is False. I changed real_request_duration invisible attribute too. Both with invisible if real_request_duration=0. I think it's a solution, but I don´t know if it´s the best solution.

Reply:
> Completion_delay is visible for everyone according to user permissions, but it is invisible when (now) end_request_datetime is False

My question is philosophical. Since `completion_delay` could show a _negative_ data for the technician, I'm not sure if this field should only be visible for a manager role... Anyway, in the `pivot` view is always visible too...



> I am unable to check end_request_datetime is False. I changed real_request_duration invisible attribute too. Both with invisible if real_request_duration=0. I think it's a solution, but I don´t know if it´s the best solution.

OK, I'll take a look for that unexpected behavior.